doc/fault_management/fault_mgmt_infra_HLD.md

```diff
@@ -0,0 +1,173 @@
+
+<img width="892" alt="Screenshot 2023-11-30 at 5 36 53 PM" src="https://github.com/shyam77git/SONiC/assets/69485234/47a5fa4a-7f3d-4d45-86a4-81cb3fb893f1">
+
+**Block diagram's markers/steps (#1 through #9) explanation:**
+|      Markers        |                   Brief Description                                    | State                                                  |
+| --------------------|------------------------------------------------------------------------|--------------------------------------------------------|
+| Markers 1 through 3 | Fault detector (source) publishes its events(faults, alarms) to eventD | Available in SONiC 202305
+| Marker 4            | Storing the events to eventD's redisDB instance                        | codeflow to be committed as part of revised HLD: https://github.com/sonic-net/SONiC/pull/1409 |                                                                            
+| Marker 5 through 9  | This is the beige oval-shaped box (on the right) representing the FM (Fault Managment) infrastructure module being added and developed as part of this HLD/Feature| This HLD|
+                                                                    
+
+# High Level Work Flow
+Following are the main functionalities/tasks of the **FM infrastructure module** being introduced as part of this document:
+1. Formulate platform/HWSKU specific Fault-Action Policy Table (json or yaml file)
+   - There would be generic (default) table if none provided by platform
+   - A platform supplied file would override the default one
+2. Introduce a new micro-service (fault_manager) at host (Linux Kernel)
+3. This service to subscribe to eventsD redisDB instance (once available) and fetch events and alarms from it
+4. Parse events against event_tag in sonic-events yang model 
+5. Analyze them against Fault-Action Policy Table (file)
+   - Take fault_type and fault_severity as input from the fetched event and perform lookup
+     on these fields in Fault-Action Policy Table to determine the action(s) needed
+6. Handle the fault (i.e. take action) based on action(s) specified in Fault-Action Policy Table
+   - action may range from logging (disk, OBFL flash etc.) to reload/shutdown etc.
+   - Taking action would either be by itself (i.e. in ts own micro-service) or delegating it to action's owner
+7. Tabulate event entry (along with action taken) for book-keeping purposes
+
+
+# Fault's End-to-End WorkFlow Sequence
+Following workflow depicts the end-to-end fault (event) flow from Fault generation to Fault Handling
+![Fault Management (FM) Workflow sequence](https://github.com/shyam77git/SONiC/assets/69485234/2b453a1b-6e14-48c6-bf61-ab978e62a3bf)
+
+
+# Fault-Action Policy Table (fault_action_policy.json)
+
+{
+
+    "chassis": {
+        "name": "PID or HWSKU",
+        "faults": [
+            {
+        
+                 "type" : "CUSTOM_EVPROFILE_CHANGE",
+                 "severity" : "MAJOR",
+                 "action" : ["syslog"]
```

Comment:
syslog is the default action, correct? do we need it as part of fault-manager action?